src/main/java/org/openrewrite/java/migrate/lang/IfElseIfConstructToSwitch.java

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright 2025 the original author or authors.
+ * <p>
+ * Licensed under the Moderne Source Available License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://docs.moderne.io/licensing/moderne-source-available-license
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate.lang;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Value;
+import org.jspecify.annotations.Nullable;
+import org.openrewrite.*;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.search.SemanticallyEqual;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.Space;
+import org.openrewrite.java.tree.Statement;
+import org.openrewrite.staticanalysis.kotlin.KotlinFileChecker;
+
+import java.time.Duration;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.openrewrite.java.migrate.lang.NullCheck.Matcher.nullCheck;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class IfElseIfConstructToSwitch extends Recipe {
+    @Override
+    public String getDisplayName() {
+        return "If-else-if-else to switch";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Replace if-else-if-else with switch statements. In order to be replaced with a switch, " +
+                "all conditions must be on the same variable and there must be at least three cases.";
+    }
+
+    @Override
+    public @Nullable Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(3);
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(Preconditions.not(new KotlinFileChecker<>()), new JavaVisitor<ExecutionContext>() {
+
+            @Override
+            public J visitIf(J.If iff, ExecutionContext ctx) {
+                SwitchCandidate switchCandidate = new SwitchCandidate(iff, getCursor());
+
+                if (switchCandidate.isValidCandidate()) {
+                    Object[] arguments = switchCandidate.buildTemplateArguments(getCursor());
+                    if (arguments.length == 0) {
+                        return iff;
```

Comment:
Are you sure you should just return the `iff` variable here? I would expected:

```suggestion
                        return super.visitIf(iff, ctx);
```